RecoTracker/MeasurementDet/plugins/MeasurementTrackerEventProducer.cc

```diff
@@ -105,6 +120,47 @@ MeasurementTrackerEventProducer::updatePixels( const edm::Event& event, PxMeasur
     }
   }
 
+  if (!theBadPixelFEDChannelsLabels.empty()) {
+    edm::Handle<PixelFEDChannelCollection> pixelFEDChannelCollectionHandle;
+    for (const edm::EDGetTokenT<PixelFEDChannelCollection>& tk: theBadPixelFEDChannelsLabels) {
+      if (!event.getByToken(tk, pixelFEDChannelCollectionHandle)) continue;
+      int i=0;
+      for (const auto& disabledChannels: *pixelFEDChannelCollectionHandle) {	
+	PxMeasurementDetSet::BadFEDChannelPositions positions;
+	for(const auto& ch: disabledChannels) {
+	  const sipixelobjects::PixelROC *roc_first=NULL, *roc_last=NULL;
+	  sipixelobjects::CablingPathToDetUnit path = {ch.fed, ch.link, 0};
+	  for (path.roc=1; path.roc<=(ch.roc_last-ch.roc_first)+1; path.roc++) {
+	    const sipixelobjects::PixelROC *roc = cablingMap.findItem(path);
+	    if (roc==NULL) continue;
+	    assert(roc->rawId()==disabledChannels.detId());
+	    if (roc->idInDetUnit()==ch.roc_first) roc_first=roc;
+	    if (roc->idInDetUnit()==ch.roc_last) roc_last=roc;
+	  }
+	  if (roc_first==NULL || roc_last==NULL) { 
+	    edm::LogError("PixelFEDChannelCollection")<<"Do not find either roc_first or roc_last in the cabling map."; 
+	    continue; 
+	  }
+	  const PixelGeomDetUnit * theGeomDet = dynamic_cast<const PixelGeomDetUnit*> (trackerGeom.idToDet(roc_first->rawId()));
+	  PixelTopology const * topology = &(theGeomDet->specificTopology());
+	  sipixelobjects::LocalPixel::RocRowCol local = {39, 25};   //corresponding to center of ROC row, col
```

Comment:
can these magic numbers (39 and 25) be derived from some available constants or the geometry/topology?